src/cache/in-memory-cache.ts

```diff
@@ -0,0 +1,84 @@
+import { LoggerInstance } from 'winston';
+import { Cache } from './cache';
+import Timer = NodeJS.Timer;
+
+export interface InMemoryCacheEntry {
+  timestamp: number;
+  data: any;
+}
+
+/**
+ * This cache implementation stores entries in a map in memory (not efficient for large caches)
+ * @example
+ * const cache = new InMemoryCache<{bar: string}>({expireAfterSeconds: 600}); // cached entries expire after 10 minutes
+ * cache.put('foo', { bar: 'baz' });
+ */
+export class InMemoryCache<T = any> implements Cache<T> {
+  private readonly keyValueStore = new Map<string, InMemoryCacheEntry>();
+  private readonly expireAfterSeconds: number;
+  private readonly logger?: LoggerInstance;
+  private cleanupIntervalRef?: Timer;
+
+  constructor({ expireAfterSeconds, logger }: { expireAfterSeconds: number, logger?: LoggerInstance }) {
+    this.expireAfterSeconds = parseFloat((expireAfterSeconds || 0).toString());
```

Comment:
Can't you set a default value for expireAfterSeconds?

Reply:
the default behavior is to never expire -- I didn't want to change this in case people were expecting it to behave in this way

Reply:
But it defaults to zero already (`(expireAfterSeconds || 0)`), I was just suggesting pushing that into the parameters... although giving it a second look, I now realise it's being spread from an object, which makes it much less feasible. Never mind :)

Reply:
right, the constructor param syntax with spreads is a bit awkward as you have to repeat everything 3 times

Reply:
You could pull the input into it's own interface time, but that is really just hiding the issue :P